### PR TITLE
Out of stock product variation not showing for configurable product when allow to display out of stock product option set to yes from backend configuration

### PR DESCRIPTION
**Pull request by abhaykhatariya**
### Description
When display out of stock product set to YES from the backend configuration,  Configurable product variation not showing.
I found this problem when configurable product have only one option like color
screenshot
screenshot 1

### Fixed Issues (if relevant)
1. magento/magento2#10454 : Attributes - Display Out of Stock Products

### Manual testing scenarios
For steps to reproduce, please check bug report : #10454 

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18184) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18184) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: abhaykhatariya<br/>:x: Abhay Khatariya<hr/>**Abhay Khatariya** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
Hi @abhaykhatariya. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by abhaykhatariya** [comment]
@magento-admin @magento-engcom-team 
Please review my commit and let me know if you need further details.

**Comment by abhaykhatariya** [comment]
@engcom-backlog-nazar 
Hello can you please review and let me know what i have to push ?

**Comment by josefbehr** [comment]
@abhaykhatariya Thanks for your contribution. Can you please fix the failing tests?

**Comment by abhaykhatariya** [comment]
@josefbehr 
Yes I will do it soon

File: app/code/Magento/ConfigurableProduct/Block/Product/View/Type/Configurable.php [hunk 1]
```diff
@@ -20,325 +21,331 @@
  * @api
  * @since 100.0.2
  */
-class Configurable extends \Magento\Catalog\Block\Product\View\AbstractView
```
Comment (sidolov) [inline app/code/Magento/ConfigurableProduct/Block/Product/View/Type/Configurable.php:23]: Please, revert code style changes for this file

File: app/code/Magento/ConfigurableProduct/Plugin/Model/ResourceModel/Attribute/InStockOptionSelectBuilder.php [hunk 2]
```diff
@@ -20,13 +21,23 @@ class InStockOptionSelectBuilder
...
+
     /**
      * @param Status $stockStatusResource
+     * @param StockConfigurationInterface $stockConfiguration
      */
-    public function __construct(Status $stockStatusResource)
-    {
+    public function __construct(
+        Status $stockStatusResource,
+        StockConfigurationInterface $stockConfiguration
```
Comment (sidolov) [inline app/code/Magento/ConfigurableProduct/Plugin/Model/ResourceModel/Attribute/InStockOptionSelectBuilder.php:37]: Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sivaschenko** [comment]
@abhaykhatariya can you please resolve the code review comments and sign the [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18184) so we can process this pull request

**Comment by abhaykhatariya** [comment]
@sivaschenko Sure will do this week and update you here

**Comment by sivaschenko** [comment]
Hi @abhaykhatariya I am closing this pull request due to inactivity. Feel free to reopen should you wish to continue progress on this contribution

**Pull request closed without merging**